realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -535,7 +535,16 @@ public String toJson() {
      */
     public boolean isValid() {
         Token userToken = getSyncUser().getUserToken();
-        return syncUser.isLoggedIn() && userToken != null && userToken.expiresMs() > System.currentTimeMillis() && SyncManager.getUserStore().isActive(syncUser.getIdentity());
+        return userToken != null && userToken.expiresMs() > System.currentTimeMillis() && !isLoggedOut();
+    }
+
+    /**
+     * Returns whether the user is logged out from the Realm Object Server or not.
```

Comment:
I feel having both `isValid` and `isLoogedOut` will confuse user a lot. I think `isLoggedOut()` doesn't have to be existing in the public API? For any cases, user should just call `isValid()` to check. Can we make this API package protected?